lib/hanami/extensions/view/slice_configured_view.rb

```diff
@@ -32,12 +32,52 @@ def inspect
         # rubocop:disable Metrics/AbcSize
         def configure_view(view_class)
           view_class.settings.each do |setting|
-            if slice.config.views.respond_to?(:"#{setting}")
-              view_class.config.public_send(
-                :"#{setting}=",
-                slice.config.views.public_send(:"#{setting}")
-              )
-            end
+            next unless slice.config.views.respond_to?(:"#{setting}")
+
+            # Configure the view from config on the slice, _unless it has already been configured by
+            # a parent slice_, and re-configuring it for this slice would make no change.
+            #
+            # In the case of most slices, its views config is likely to be the same as its parent
+            # (since each slice copies its `config` from its parent), and if we re-apply the config
+            # here, then it may possibly overwrite config customisations explicitly made in parent
+            # view classes.
+            #
+            # For example, given an app-level base view class, with custom config:
+            #
+            #   module MyApp
+            #     class View < Hanami::View
+            #       config.layout = "custom_layout"
+            #     end
+            #   end
+            #
+            # And then a view in a slice inheriting from it:
+            #
+            #   module MySlice
+            #     module Views
+            #       class SomeView < MyApp::View
+            #       end
+            #     end
+            #   end
+            #
+            # In this case, `SliceConfiguredView` will be extended two times:
+            #
+            # 1. When `MyApp::View` is defined
+            # 2. Again when `MySlice::Views::SomeView` is defined
+            #
+            # If we blindly re-configure all view settings each time `SliceConfiguredView` is
+            # extended, then at the point of (2) above, we'd end up overwriting the custom
+            # `config.layout` explicitly configured in the `MyApp::View` base class, leaving
+            # `MySlice::Views::SomeView` with `config.layout` of `"app"` (the default as specified
+            # at `Hanami.app.config.views.layout`), and not the `"custom_layout"` value configured
+            # in its immediate superclass.
+            #
+            # This would be surprising behavior, and we want to avoid it.
+            slice_value = slice.config.views.public_send(:"#{setting}")
+            parent_value = slice.parent.config.views.public_send(:"#{setting}") if slice.parent
```

Comment:
Can’t we rely on `Config#[]` here?

Reply:
Unfortunately not, since `config.views` here is and instance of `Hanami::Configuration::Views`, which delegates to an inner `base_configuration` object for most of its settings. I could probably extend it to add support for delegating to that object for `#[]`, but for right now I might leave it as is (that kind of change feels a little outside of the scope of this bug fix PR).